src/huggingface_hub/hf_api.py

```diff
@@ -9525,6 +9529,51 @@ def list_user_following(self, username: str) -> Iterable[User]:
         for followed_user in r.json():
             yield User(**followed_user)
 
+    def auth_check(self, repo_id: str, token: Union[bool, str, None] = None) -> None:
```

Comment:
I think it would be more convenient if the method returns  `True` when the user has access. This change would allow users to use the boolean return in conditional statements

Reply:
Agree!